File: my-frontend/src/types/assistant.ts
import type { GraphEdge, GraphNode } from '@/types/graph';

export type AssistantRole = 'user' | 'assistant' | 'system';

export interface AssistantMessage {
  id: string;
  role: AssistantRole;
  content: string;
  createdAt: string;
}

/** 后端持久化的会话（不含消息体，消息单独拉取）。 */
export interface Conversation {
  id: string;
  title: string | null;
  createdAt: string;
  updatedAt: string;
}

/** 后端持久化的一条消息。 */
export interface BackendMessage {
  id: string;
  conversationId: string;
  role: AssistantRole;
  content: string;
  createdAt: string;
}

/** 前端展示用会话：后端 Conversation + 本地消息缓存。 */
export interface AssistantSession {
  id: string;
  title: string;
  messages: AssistantMessage[];
  createdAt: string;
  updatedAt: string;
}

/**
 * 后端 build_graph 意图产出的图谱草稿（SSE graph_draft 事件）。
 * 节点不含 x/y，由前端落库时布局；边引用的节点 id 已保证在 nodes 内。
 */
export interface GraphDraft {
  title: string;
  nodes: Omit<GraphNode, 'x' | 'y'>[];
  edges: GraphEdge[];
}

File: backend/app/schemas/assistant.py
from pydantic import BaseModel, ConfigDict, Field


class Conversation(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    title: str | None = None
    created_at: str = Field(alias="createdAt")
    updated_at: str = Field(alias="updatedAt")


class Message(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    conversation_id: str = Field(alias="conversationId")
    role: str
    content: str
    created_at: str = Field(alias="createdAt")


class ChatRequest(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    conversation_id: str = Field(alias="conversationId")
    content: str = Field(min_length=1)
